api/v1alpha1/etcdpeer_types.go

```diff
@@ -7,10 +7,52 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+// InitialClusterMemeber describes a single member of the initial cluster.
+type InitialClusterMember struct {
+	// Name is a friendly name for the peer, used as a means to identify the
+	// peer once it has joined a cluster. This should match the `name` field
+	// of the `EtcdPeer` resource representing that peer.
+	Name string `json:"name"`
+
+	// Host forms part of the Advertise URL - the URL at which this peer can
+	// be contacted. The port and scheme are hardcoded to 2380 and http
+	// respectively.
+	Host string `json:"host"`
+}
+
+// StaticBootstrap provides static contact information for initial members of
+// the cluster.
+type StaticBootstrap struct {
+	// InitialCluster provides details of all initial cluster members,
+	// and should include ourselves.
+	// +kubebuilder:validation:MinItems:=1
+	InitialCluster []InitialClusterMember `json:"initalCluster,omitempty"`
+}
+
+// Bootstrap contains bootstrap infromation for the peer to use.
+type Bootstrap struct {
+	// Static boostrapping requires that we know the network names of the
+	// other peers ahead of time.
+	// +optional
+	Static StaticBootstrap `json:"static,omitempty"`
+}
+
 // EtcdPeerSpec defines the desired state of EtcdPeer
 type EtcdPeerSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
+
+	// The name of the etcd cluster that this peer should join. This will be
+	// used to set the `spec.subdomain` field and the
+	// `etcd.improbable.io/cluster-name` label on the Pod running etcd.
+	// +kubebuilder:validation:MaxLength:=64
+	ClusterName string `json:"clusterName"`
+
+	// Bootstrap is the bootstrap configuration to pass down into the etcd
+	// pods. As per the etcd documentation, etcd will ignore bootstrap
+	// instructions if it already knows where it's peers are.
+	// +optional
+	Bootstrap Bootstrap `json:"bootstrap,omitempty"`
```

Comment:
Is this struct actually optional?

If yes - can you make `Bootstrap` a pointer here as well.
If no - can you remove `omitempty` and `+optional` tags?

Reply:
Resolved in https://github.com/improbable-eng/etcd-cluster-operator/pull/19